galaxy_ng/app/access_control/access_policy.py

```diff
@@ -213,6 +243,50 @@ def has_distribution_repo_perms(self, request, view, action, permission):
 
         return False
 
+    def signatures_not_required_for_repo(self, request, view, action):
+        """
+        Validate that collections are being added with signatures to approved repos
+        when signatures are required.
+        """
+        repo = view.get_object()
+        repo_version = repo.latest_version()
+
+        if settings.GALAXY_REQUIRE_SIGNATURE_FOR_APPROVAL:
+            return True
+
+        serializer = CollectionVersionCopyMoveSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        data = serializer.validated_data
+
+        signing_service = data.get("signing_service", None)
+
+        if signing_service:
+            return True
+
+        is_any_approved = False
+        for repo in data["destination_repositories"]:
+            if repo.pulp_labels.get("pipeline", None) == "approved":
+                is_any_approved = True
+                break
+
+        # If any destination repo is marked as approved, check that the signatures
+        # are available
+        if not is_any_approved:
+            return True
+
+        for cv in data["collection_versions"]:
+            sig_exists = repo_version.get_content(
+                ansible_models.CollectionVersionSignature.objects
+            ).filter(signed_collection=cv).exists()
+
+            if not sig_exists:
+                raise ValidationError(detail={"collection_versions": _(
+                    "Signatures are required in order to add collections into any 'approved'"
+                    "repository when GALAXY_REQUIRE_SIGNATURE_FOR_APPROVAL is enabled."
+                )})
```

Comment:
on 254 it checks for `settings.GALAXY_REQUIRE_SIGNATURE_FOR_APPROVAL` and early returns `True` 
but in line 283 it raises error when that setting is enables, is that intended?

Reply:
that's wrong... why didn't my tests catch this...